.github/workflows/main.yml

```diff
@@ -9,57 +14,58 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4
-      - uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
         with:
-          python-version: '3.14.0'
+          enable-cache: false
+          ignore-empty-workdir: true
+
       - name: create project
         run: |
-          pip install django
-          django-admin startproject \
-           --template="https://github.com/lincolnloop/django-layout/zipball/$GITHUB_REF_NAME" \
+          uv run --with django django-admin startproject \
+           --template="$TEMPLATE_URL" \
            --extension=py,md,gitignore,yaml,json,toml \
            --name=Makefile,Dockerfile \
            --exclude=.github \
-           testproj
+           testproj .
+
       - name: mock client build
         run: |
           mkdir -p client/dist
           touch client/dist/index.js
-        working-directory: testproj
+
+      - name: build container
+        run: docker compose build
+
       - name: start services
-        run: |
-          docker compose up -d
-          # wait for migrations to finish
-          docker attach $(docker compose ps -q django-migrate)
+        run: docker compose up -d db client
+
+      - name: run migrations
+        run: docker compose run app python manage.py migrate --no-input
+
+      - name: start django
+        run: docker compose up -d app
         timeout-minutes: 5
-        working-directory: testproj
+
       - name: verify services running
         run: |
           docker ps  # show service status for debugging
           test $(docker ps -q | wc -l) = "3"
       - name: verify website running
         run: |
-          curl -I --retry 10 --retry-all-errors --fail localhost:8000/testproj-admin/login/
+          curl -I --retry 10 --retry-all-errors --fail \
+          localhost:8000/testproj-admin/login/
       - name: verify client build
         run: |
-          curl -I --retry 10 --retry-all-errors --fail localhost:8000/static/index.js
+          curl -I --retry 10 --retry-all-errors --fail \
+          localhost:8000/static/index.js
       - name: run tests
         run: docker compose run --rm app make test
-        working-directory: testproj
       - name: verify README.md
         run: docker compose run --rm --no-deps app cog --check README.md
-        working-directory: testproj
-      - name: lint code
-        run: docker compose run --rm --no-deps app make lint
-        working-directory: testproj
-      - name: format code
-        run: docker compose run --rm --no-deps app ruff format --check .
-        working-directory: testproj
```

Comment:
It's possible that after template substitution that the project isn't compliant anymore. Is it worth running a pre-commit check here?

Reply:
Great point – hadn't considered this. Let me try adding a second job to pre-commit; the existing one clones the repo. The new one would install via `startproject` and run on the "built" source.